demo/DemoArticles/update.js

```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/env node
+/* eslint-disable require-jsdoc */
+
+const fs = require('fs')
+const request = require('request')
+const indexJSON = require('./articles.json')
+const ArticleRef = require('./ArticleRef.js').ArticleRef
+const ArticleRefSourceType = require('./ArticleRef.js').ArticleRefSourceType
+const flattenArrayOfArrays = require('../DemoUtilities.js').flattenArrayOfArrays
+
+const DATA_PATH = './data/'
+
+const articleRefs = flattenArrayOfArrays(
+  indexJSON.map(articleData => [
+    new ArticleRef(
+      articleData.lang,
+      articleData.title,
+      articleData.revision,
+      ArticleRefSourceType.mobileView
```

Comment:
> Please however consider that ArticleRefSourceType can easily become outdated when a new type is added and new states aren't added to every place it's referenced.

I *think* it's still easier to add them this way because all of the client logic is type agnostic (with exception of creation of new ArticleRef's).